src/stream_decoder.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * File:   stream_decoder.h
+ * Author: Simon N
+ */
+
+/**
+ * @defgroup stream Stream Decoder
+ * @brief Unpack messages from the host.
+ *
+ * @addtogroup stream
+ * @{
+ * @file stream_decoder.h
+ * @brief Unpack messages from the host.
+ */
+
+#ifndef SRC_STREAM_DECODER_H_
+#define SRC_STREAM_DECODER_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+  * @brief A de-serialized message.
+  */
+typedef struct {
+  uint16_t command;  //!< The Command
+  uint16_t length;   //!< The payload length
+  const uint8_t* payload;  //!< A pointer to the payload data.
+} Message;
+
+
+/**
+ * @brief A function pointer used to handle new messages.
+ * @param message The message to handle.
+ */
+typedef void (*MessageHandler)(const Message*);
+
+
+/**
+ * @brief Initialize the Stream Decoder.
+ * @param handler The MessageHandler to call with each new message.
```

Comment:
It just seems nicer to me when the input maps more to the output doxygen, so all the info, then the params.